pages/api/id/[form]/submission/confirm.ts

```diff
@@ -0,0 +1,158 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { logMessage } from "@lib/logger";
+import { middleware, cors, jsonValidator, sessionExists } from "@lib/middleware";
+import uuidArraySchema from "@lib/middleware/schemas/uuid-array.schema.json";
+import {
+  DynamoDBDocumentClient,
+  BatchGetCommand,
+  TransactWriteCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { MiddlewareProps, WithRequired } from "@lib/types";
+import { connectToDynamo } from "@pages/api/dynamodb-client-helper";
+import { checkOne } from "@lib/cache/flags";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse, props: MiddlewareProps) => {
```

Comment:
Since we know for sure that `session` will not be undefined because we're checking with the `sessionExists()` middleware we can do a stonger typing with: `WithRequired<MiddlewareProps, "session">`.
This way session is already determined to be `Session` and not `Session | undefined` and we can remove line 10 `if (!session) .....`

Reply:
While reviewing https://github.com/cds-snc/platform-forms-client/pull/1607 I noticed that you were using `WithRequired` so I changed it on mine. I guess you did review it just before :)